src/modifierCreepingDeath.ts

```diff
@@ -0,0 +1,52 @@
+import { registerEvents, registerModifiers } from "./cards";
+import { getOrInitModifier } from "./cards/util";
+import * as Unit from './entity/Unit';
+import * as config from './config';
+import * as colors from './graphics/ui/colors';
+import Underworld from './Underworld';
+import { chooseOneOfSeeded, getUniqueSeedString, randInt } from "./jmath/rand";
+import { UnitSubType } from "./types/commonTypes";
+import { suffocateCardId } from "./cards/suffocate";
+import seedrandom from "seedrandom";
+import { makeManaTrail } from "./graphics/Particles";
+
+// Inflicts a random enemy with [quantity] stacks of suffocate on kill
+export const creepingDeathId = 'Creeping Death';
+export default function registerCreepingDeath() {
+  registerModifiers(creepingDeathId, {
+    description: i18n('rune_creeping_death'),
+    costPerUpgrade: 100,
+    add: (unit: Unit.IUnit, underworld: Underworld, prediction: boolean, quantity: number = 1) => {
+      getOrInitModifier(unit, creepingDeathId, { isCurse: false, quantity, keepOnDeath: false }, () => {
+        Unit.addEvent(unit, creepingDeathId);
+      });
+    }
+  });
+  registerEvents(creepingDeathId, {
+    onKill: async (unit: Unit.IUnit, killedUnit: Unit.IUnit, underworld: Underworld, prediction: boolean) => {
+      const modifier = unit.modifiers[creepingDeathId];
+      if (modifier) {
+        if (killedUnit) {
+          let units = prediction ? underworld.unitsPrediction : underworld.units;
+          // Find a random enemy unit and suffocate it
+          // Unit must be alive, in enemy faction, not a doodad, and not already suffocating
```

Comment:
Why not allow it to add to the stacks of already suffocating units?

Reply:
This was initially an onTurnStart effect and it felt like bad rng when it would reapply suffocate to things that were already suffocating, but now that it's onKill and it can chain... that doesn't necessarily sound like a bad thing... Yeah, we could have it add to existing stacks. It would feel better in situations where the whole map is already suffocated too, I'll update this